src/app/stores/app_store.ts

```diff
@@ -223,6 +224,51 @@ export class AppStore extends BaseStore {
     this.loadState(state);
   }
 
+  // removes unused scale objecs
+  private updateChartState() {
+    function hasMappedProperty(
+      mappings: Specification.Mappings,
+      scaleId: string
+    ) {
+      for (const map in mappings) {
+        if (mappings[map].type === "scale") {
+          if ((mappings[map] as any).scale === scaleId) {
+            return true;
+          }
+        }
+      }
+      return false;
+    }
+    const chart = this.chart;
+
+    function scaleFilter(scale: any) {
+      return !(
+        chart.elements.find((element: any) => {
+          if ((element as Specification.Object).mappings) {
+            const mappings = (element as Specification.Object).mappings;
```

Comment:
You could move this variable outside of the `if` statement, and do `if (mappings)`...so

```
const mappings = (element as Specification.Object).mappings;
if (mappings) {
    return hasMappedProperty(mappings, scale._id);
}
```